src/affine_dimensions.jl

```diff
@@ -0,0 +1,456 @@
+    uconvert(qout, uexpand(qin))
+end
+
+function uconvert(qout::UnionAbstractQuantity{<:Any,<:AbstractAffineDimensions}, qin::AbstractQuantityOrArray{<:Any,<:AbstractSymbolicDimensions})
+    uconvert(qout, uexpand(qin))
+end
+
+function uconvert(qout::UnionAbstractQuantity{<:Any,<:AbstractAffineDimensions}, qin::AbstractQuantityOrArray{<:Any,<:AbstractAffineDimensions})
+    uconvert(qout, uexpand(qin))
+end
+
+#This is needed for parsing out (*,/) in string macros 
+for (op, combine) in ((:+, :*), (:-, :/))
+    @eval function map_dimensions(op::typeof($op), args::AffineDimensions...)
+        map(assert_no_offset, args)
+        return AffineDimensions(
+            scale=($combine)(map(uscale, args)...),
+            offset=zero(Float64),
+            basedim=map_dimensions(op, map(ubasedim, args)...) 
+        )
+    end
+end
+
+#This is required because /(x::Number) results in an error, so it needs to be cased out to inv
+function map_dimensions(op::typeof(-), d::AffineDimensions) 
+    assert_no_offset(d)
+    return AffineDimensions(
+        scale=inv(uscale(d)),
+        offset=zero(Float64),
+        basedim=map_dimensions(op, ubasedim(d))
+    )
+end
+
+#This is needed for parsing out exponentials in string macros
+function map_dimensions(fix1::Base.Fix1{typeof(*)}, l::AffineDimensions{R}) where {R}
+    assert_no_offset(l)
+    return AffineDimensions(
+        scale=uscale(l)^fix1.x,
+        offset=zero(Float64),
+        basedim=map_dimensions(fix1, ubasedim(l))
+    )
+end
+
+#Generic fallback (slower and less accurate but works for more cases)
+function map_dimensions(op::F, args::AffineDimensions...) where {F<:Function}
+    assert_no_offset.(args)
+    return AffineDimensions(
+        scale=exp(op(log.(uscale.(args))...)),
+        offset=zero(Float64),
+        basedim=map_dimensions(op, ubasedim.(args)...)
+    )
+end
+
+# This function works like uexpand but will throw an error if the offset is 0
+function _no_offset_expand(q::Q) where {T,R,D<:AbstractAffineDimensions{R},Q<:UnionAbstractQuantity{T,D}}
+    return convert(with_type_parameters(Q, T, Dimensions{R}), q)
+end
+
+# Addition will return Quantity{T, Dimensions}
+Base.:(+)(q1::UnionAbstractQuantity{<:Any,<:AffineDimensions}, q2::UnionAbstractQuantity{<:Any,<:AffineDimensions}) = _no_offset_expand(q1) + _no_offset_expand(q2)
+
+# Subtraction will return Quantity{T, Dimensions}, in special cases, differences between offsetted AffineDimensions is allowed as offsets cancel out
+function Base.:(-)(q1::UnionAbstractQuantity{<:Any,<:AffineDimensions}, q2::UnionAbstractQuantity{<:Any,<:AffineDimensions})
+    if dimension(q1) == dimension(q2)
+        return uexpand(q1) - uexpand(q2)
+    else
+        return _no_offset_expand(q1) - _no_offset_expand(q2)
+    end
+end
+
+for op in (:(==), :(≈))
+    @eval Base.$op(q1::UnionAbstractQuantity{<:Any,<:AffineDimensions}, q2::UnionAbstractQuantity{<:Any,<:AffineDimensions})   = $op(uexpand(q1), uexpand(q2))
```

Comment:
Are these needed? I would have thought the promotion rules would already get it to work